src/components/SQForm/__tests__/SQFormDropdown.test.js

```diff
@@ -0,0 +1,202 @@
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox', /dropdown example/i);
+
+  expect(optionsList).toBeInTheDocument();
+  expect(optionsList.children.length).toBe(3);
+});
+
+it('should render with initial value', () => {
+  const initialValues = {
+    dropdownExample: 2
+  };
+
+  const renderDropdown = createDropdown();
+
+  renderSQForm(renderDropdown, {initialValues});
+
+  const labelValue = screen.getByRole('button', /dropdown example/i);
+  expect(labelValue).toHaveTextContent('Test 2');
+});
+
+it('should update when option is selected', () => {
+  const handleChange = jest.fn();
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    onChange: handleChange
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const option = screen.getByText('Test 1');
+  userEvent.click(option);
+
+  expect(handleChange).toHaveBeenCalled();
+  expect(handleChange).toHaveBeenCalledTimes(1);
+
+  const labelValue = screen.getByRole('button', /dropdown example/i);
+  expect(labelValue).toHaveTextContent('Test 1');
+});
+
+it('should disable option', () => {
+  const options = [
+    {label: 'Test 1', value: 1},
+    {label: 'Test 2', value: 2, isDisabled: true},
+    {label: 'Test 3', value: 3}
+  ];
+
+  const renderDropdown = createDropdown(options, DEFAULT_PROPS);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox', /dropdown example/i);
+
+  expect(optionsList.childNodes[1]).toHaveAttribute('aria-disabled', 'true');
+});
+
+it('should display an empty option when displayEmpty is true', () => {
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    displayEmpty: true
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox', /dropdown example/i);
+
+  expect(optionsList.children.length).toBe(4);
+  expect(optionsList.childNodes[0]).toHaveTextContent('- -');
+});
+
+it('should not be selectable if it is disabled', () => {
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    isDisabled: true
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  expect(expandButton).toHaveAttribute('aria-disabled', 'true');
```

Comment:
Does the presence of `aria-disabled` ensure the element is not clickable?

My nooby brain expected something like `expect(handleChange).toHaveBeenCalledTimes(0);`